File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    # Validate input
    if not isinstance(number, int) or number <= 0:
        raise ValueError("Classification is only possible for positive integers.")
    
    # Calculate aliquot sum (sum of proper divisors)
    aliquot_sum = 0
    for i in range(1, number):
        if number % i == 0:
            aliquot_sum += i
    
    # Classify based on aliquot sum
    if aliquot_sum == number:
        return "perfect"
    elif aliquot_sum > number:
        return "abundant"
    else:
        return "deficient"

File: solutions/javascript/perfect-numbers/1/perfect-numbers.js
export const classify = (num) => {
  if (num <= 0) {
    throw new Error('Classification is only possible for natural numbers.');
  }

  let aliquotSum = 0;

  for (let i = 1; i * i <= num; i++) {
    if (num % i === 0) {
      if (i !== num) {
        aliquotSum += i;
      }
      const j = num / i;
      if (j !== i && j !== num) {
        aliquotSum += j;
      }
    }
  }

  if (aliquotSum === num) return 'perfect';
  if (aliquotSum > num) return 'abundant';
  return 'deficient';
};